### ISSUE TITLE
Unable to install module "auth_ldap" because an external dependency is not met: No module named ldap

### ISSUE THREAD

**Issue by youmingzhao** [odoo/docker#140]
Hi all;
I have pull odoo 11 from http://hub.docker.com; when i start the odoo docker container. after that ,i want to setup  "authentication via ldap" module. it gives an error:

Odoo Server Error - Warning
Unable to install module "auth_ldap" because an external dependency is not met: No module named ldap

after that i find the answer as "https://www.odoo.com/fr_FR/forum/aide-1/question/odoo-11-ldap-authentication-125131"

but i can not exec the command in docker container's command:
$ sudo apt install libldap2-dev libsasl2-dev
$ sudo pip3 install pyldap
it say:
bash: $: command not found

could you please help me ,thanks very much

Yours youming.zhao

**Comment by nemmeviu** [comment]
For ldap suport you can use this command:
$ docker exec -u 0 -ti container_id bash
# pip3 install pyldap

----
maybe we can make one fork of odoo project and add ldap support.
I'm try

**Comment by youmingzhao** [comment]
@nemmeviu , thank you very much,it was solved . thank you again.

**Comment by thomas15v** [comment]
Not recommended to do it this way, if you update the image those chances will be gone. It's better to build the image with those commands:

**DockerFile:**
```DockerFile
FROM odoo:11

USER root

RUN apt-get update && apt-get install -y --no-install-recommends python3-dev gcc libldap2-dev libssl-dev libsasl2-dev
RUN pip3 install pyldap

USER odoo
```
To use this DockerFile you can use the following docker-compose file.
**docker-compose.yml:**
```yml
version: '2'
services:
  db:
    image: postgres:9.4
    restart: always
    environment:
      - POSTGRES_USER=odoo
      - POSTGRES_PASSWORD=odoo
  odoo:
    build: .
    restart: always
    links:
      - db
    ports:
      - 80:8069
```

**Comment by nemmeviu** [comment]
@thomas15v look this: https://github.com/odoo/docker/pull/149

If odoo approve, and uncomment the lines, ldap support pass to native image.

**Comment by eugene-bright** [comment]
You can use my image `eugenebright/odoo:11.0` as a temporary solution.

**Comment by eugene-bright** [comment]
Looks like it's fixed in 05c776bc.
Please check fresh upstream images.

**Comment by d-fence** [comment]
Already fixed in 05c776b
